src/clapsolver/clap.py

```diff
@@ -33,3 +33,124 @@ def cost(i, j, costs):
 
     # Total cost is that of the submatrix lsap plus the cost of the constraint.
     return sub_costs[sub_row_ind, sub_col_ind].sum() + costs[i, j]
+
+
+def constrained_lsap_costs(costs):
+    """Solve a constrained linear sum assignment problem for each entry.
+
+    TODO: More thorough testing of this function.
```

Comment:
Once tests have been written, remove this TODO comment.

Reply:
I created a test using the cost matrices given. Is this test sufficient?